code/modules/atmospherics/machinery/pipes/smart.dm

```diff
@@ -12,69 +12,83 @@ GLOBAL_LIST_INIT(atmos_components, typecacheof(list(/obj/machinery/atmospherics)
 	icon = 'icons/obj/atmospherics/pipes/pipes_bitmask.dmi'
-	connections = NONE
 
+	//find all directions this pipe is connected with other nodes
+	connections = NONE
 	for(var/i in 1 to device_type)
 		if(!nodes[i])
 			continue
 		var/obj/machinery/atmospherics/node = nodes[i]
 		var/connected_dir = get_dir(src, node)
 		connections |= connected_dir
-	var/bitfield = CARDINAL_TO_FULLPIPES(connections)
-	dir = check_binary_direction(connections)
-
-	// If we dont have enough bits to make a proper sprite, add some shortpipe bits
+	//set the correct direction for this node in case of binary directions
+	switch(connections)
+		if(EAST | WEST)
+			dir = EAST
+		if(SOUTH | NORTH)
+			dir = SOUTH
+		else
+			dir = connections
 
-	// Smart pipe icons differ from classic pipe icons in that we stop adding
-	// short pipe directions as soon as we find a valid sprite, rather than
-	// adding in all connectable directions.
-	// This prevents a lot of visual clutter, though it does make it harder to
-	// notice completely disconnected pipes.
-	if(ISSTUB(connections))
-		var/bits_to_add = NONE
-		if(connections != NONE)
-			bits_to_add |= REVERSE_DIR(connections) & initialize_directions
-		var/candidates = initialize_directions
-		var/shift = 0
-		// Note that candidates "should" never reach 0, as stub pipes are not allowed and break things
-		while (ISSTUB(connections | bits_to_add) && (candidates >> shift) != 0)
-			bits_to_add |= candidates & (1 << shift)
-			shift += 1
-		bitfield |= CARDINAL_TO_SHORTPIPES(bits_to_add)
+	//same as connections but used for spriting
+	var/sprite_bits = NONE
+	//the directions this pipe stretches out in e.g. T pipe is EAST,WEST & SOUTH, L pipe is NORTH,EAST & so on
+	var/list/spanning_directions = get_node_connects()
+	/**
+	 *For pipes created during mapload we draw the pipes sprite only in directions where its connected to a machine
+	 *so for example if an T shaped pipe is connected only in its EAST & WEST directions then only those ends are drawn
+	 *but the SOUTH end is not drawn
+	 *this will allow mappers to use whatever pipes but the end result has no visual clutter.
+	 *This is actually just an bandage for lazy mappers using + pipes all over the place without carying about directions so hopefully when they map pipes correctly we can remove this
+	 */
+	if(map_loaded_pipe)
+		sprite_bits = connections
+		/**
+		 * if pipe is connected in only one direction[e.g. after disconnecting its neighbour] then to avoid a broken sprite append the reverse direction of its one connected end.
+		 * this wont work for L pipes because if one of its ends is broken then the opposite direction of any of its last connected end is invalid
+		 * e.g. for an L pipe if the top[NORTH] end is broken the opposite of its one remaining connected end[i.e EAST END] is WEST but thats not an valid direction for this pipe
+		 * so we have to again check one last time after this to make sure the pipe isnt broken
+		 */
```

Comment:
nice docs.

Reply:
Thanks.  Gotta tell the full story u know